core/src/main/scala/org/http4s/MessageFailure.scala

```diff
@@ -77,6 +78,10 @@ object ParseResult {
       case NonFatal(e) => Left(ParseFailure(sanitized, e.getMessage))
     }
 
+  private[http4s] def fromParser[A](parser: Parser[A], errorMessage: => String)(
+      s: String): ParseResult[A] =
+    parser.parseAll(s).leftMap(e => ParseFailure(errorMessage, e.toString))
```

Comment:
Yep, this was what I saw in the other places this pattern was used. Will investigate.

Reply:
So, what we get out of `Parser.Error` is an offset where the parsing failed, and a list of expected inputs that it didn't find. I can format that in a decently nice way, but it's not going be as good as the ErrorFormatter from parboiled.

An alternative is to do something like https://github.com/http4s/http4s/blob/main/core/src/main/scala/org/http4s/HttpVersion.scala#L58 and repeat the input instead of giving parser-specific errors.

Reply:
The new `LocationMap` in cats-parse helps turn input into a row-column index and is designed to help with formatting. Oscar mentioned some code in bosatsu that formats errors, but I haven't gone hunting for it.

I suspect there's an opportunity to do a decent default rendering in cats-parse on that `LocationMap`.